core/trino-main/src/main/java/io/trino/execution/TaskStateMachine.java

```diff
@@ -39,12 +46,20 @@
     private final DateTime createdTime = DateTime.now();
 
     private final TaskId taskId;
+    private final Executor executor;
     private final StateMachine<TaskState> taskState;
     private final LinkedBlockingQueue<Throwable> failureCauses = new LinkedBlockingQueue<>();
 
+    private final Object lock = new Object();
+    @GuardedBy("lock")
```

Comment:
this?

Reply:
answered here: https://github.com/trinodb/trino/pull/9361#discussion_r751195606

Reply:
The comment you linked does not apply here. There is no code synchronized on `this` here so far. All the synchronized sections use `lock`.

Reply:
Removing "lock" object, synchronizing on "this".